source/content.js

```diff
@@ -65,6 +81,98 @@ function removeProfileHeader() {
 	$('.ProfileCanopy-header').remove();
 }
 
+function onDMModalOpenAndClose(handleConversationOpen, handleDMModalClose) {
+	observeEl('#dm_dialog', mutations => {
+		for (const mutation of mutations) {
+			if (mutation.target.style.display === 'none') {
+				// This is necessary for `handleMessageChange` function
+				setToLocalStorage({isDMModalOpen: false});
```

Comment:
The message in the contenteditable div (`#tweet-box-dm-conversation`) gets reset on DM close, which means that `onMessageChange` function gets fired ( because of mutation observer on that contenteditable div). So every time the DM closes, the last opened conversation's message is set to an empty message. Instead of an event emitter way to tell `onMessageChange` that DM has closed and that it shouldn't run the local state set to an empty message, I did it with setting it on local storage. I hope I'm being clear in what I did? If not, I'll try again.

Also I couldn't find a proper/correct way to do this, so I did this bit hackish thing to make it work. I'd really like to be suggested how this could be bettered.

Reply:
@sindresorhus